pkg/reconcile/pipeline/handler/project/impl.go

```diff
@@ -70,6 +70,11 @@ func BindingsAsEnv(ctx pipeline.Context) {
 		return
 	}
 
+	if len(ctx.BindingItems()) == 0 {
```

Comment:
we should put this check in `PreFlightCheck` function.

Reply:
+1